pennylane_lightning/core/_measurements_base.py

```diff
@@ -215,14 +215,15 @@ def get_measurement_function(
             if isinstance(measurementprocess, ProbabilityMP):
                 return self.probs
 
-            if isinstance(measurementprocess, VarianceMP):
+            if isinstance(measurementprocess, VarianceMP) and measurementprocess.obs is not None:
```

Comment:
Do we have any tests for `expval` and `var` without `obs`?

Reply:
These will be covered by tests that use `expval/var(mcm_expression)`